### PR TITLE
Added text editor within VisualGit 

### PR DESCRIPTION
**Pull request by cli727**
- The panel that displays insertions/deletions of modified files now displays the file content and allows the content to be edited when 'Edit' is pressed
- 'Save' and 'Cancel' buttons are created in a new div. Allows saving or cancel the editing of the modified file
- A new script 'file.ts' is added to deal with file reading and saving, and the corresponding behavior when 'Save' and 'Cancel' buttons are pressed; This file is also added in 'index.html'
- 'router.ts' has been modified to deal with hiding/displaying the 'Save' and 'Cancel' buttons when their parent panel is hidden/displayed
- 'index.html' has been modified to use 'file.ts'
- 'style.css' has been modified to style 'Save' and 'Cancel' buttons
- Closes issue #32

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -78,7 +88,14 @@
...
              console.log(repoFullPath + "/" + content);
-             opn(repoFullPath + "/" + content);
+             let doc = document.getElementById("diff-panel-body");
+             doc.innerHTML = "";
+             doc.contentEditable = "true";
+
+             displayDiffPanel();
+             enableSaveCancelButton();
+             readFromFile(content);
+             //opn(repoFullPath + "/" + content);
```
Comment (EmilyYVChan) [inline index.html]: Is this meant to stay commented out?

Comment (cli727) [reply index.html]: Yeah it won't affect the functionality. But I will remove it for good practice sake : )

**Review by EmilyYVChan — CHANGES_REQUESTED** [review]
I found a bug where the user is able to edit the text in the file although the user is not in 'Edit' mode.
To reproduce this failure:

Attempt to edit text in a file. But instead of saving the changes, click cancel.
Click on the file to open the diff panel.
Type directly into the diff panel. This is where the bug is.
@cli727 kindly amend.

**Comment by cli727** [comment]
@EmilyYVChan Thanks, that was indeed a bug. The html element was simply forgotten to be set non-editable after user cancels the operation. The bug should be fixed with the latest commit and please kindly review again.

**Review by EmilyYVChan — APPROVED** [review]
Have tested the bug that Chen fixed. Looks good now.

**Comment by EmilyYVChan** [comment]
@softeng-701 ready for merge :)

**Comment by softeng-701** [comment]
@EmilyYVChan please squash your commits

**Comment by EmilyYVChan** [comment]
@cli727 I think it's better if you do the squashing since you are the primary owner of this issue.

**Comment by cli727** [comment]
@softeng-701 Changes from upstream are pulled and merged; all commits squashed; ready for merge

**Pull request merged**